teuthology/orchestra/test/test_systemd.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import os
+
+from teuthology.orchestra import cluster
+from teuthology import misc
+from teuthology.orchestra.daemon.group import DaemonGroup
+import subprocess
+
+
+class FakeRemote(object):
+    pass
+
+
+def test_pid():
+    ctx = argparse.Namespace()
+    ctx.daemons = DaemonGroup(use_systemd=True)
+    remote = FakeRemote()
+
+    file_path = os.path.join(os.path.dirname(__file__),
+                             "files/daemon-systemdstate-pid-ps-ef.output")
+
+    with open(file_path) as ps:
+        ps_output = ps.read()
+    print("\nps -ef commond output: \n%s" % ps_output)
+
+    def sh(args):
+        args[0] = "echo"
+        args[1] = '"{}"'.format(ps_output)
+        print(args[2:])
+        args = " ".join([arg.value if hasattr(arg, "value")
```

Comment:
Since we are doing subprocess call instead testing pure python regex, we should isolate test code from the escaping the `ps -ef` output, so instead of 'echo' can we just add:
```
args[0] = 'cat'
args[1] = ps_ef_output_path
```
So we don't need to read it.

Reply:
and this can be done with single assignment: args[0:2] = ['cat', ps_ef_output_path]

Reply:
Instead of inventing own way of escaping/converting list to a string, wouldn't it be better to use `teuthology.orchestra.run.quote`?

Reply:
updated.